aspnetcore/migration/1x-to-2x/identity-2x.md

```diff
@@ -0,0 +1,327 @@
+    ```
+
+### Microsoft Account Authentication
+Make the following changes in *Startup.cs*:
+- Replace the `UseMicrosoftAccountAuthentication` method call in the `Configure` method with `UseAuthentication`:
+
+    ```csharp
+    app.UseAuthentication();
+    ```
+
+- Invoke the `AddMicrosoftAccount` method in the `ConfigureServices` method:
+
+    ```csharp
+    services.AddAuthentication().AddMicrosoftAccount(options => {
+        options.ClientId = Configuration["auth:microsoft:clientid"];
+        options.ClientSecret = Configuration["auth:microsoft:clientsecret"];
+    });
+    ``` 
+
+### Twitter Authentication
+Make the following changes in *Startup.cs*:
+- Replace the `UseTwitterAuthentication` method call in the `Configure` method with `UseAuthentication`:
+ 
+    ```csharp
+    app.UseAuthentication();
+    ```
+
+- Invoke the `AddTwitter` method in the `ConfigureServices` method:
+
+    ```csharp
+    services.AddAuthentication().AddTwitter(options => {
+        options.ConsumerKey = Configuration["auth:twitter:consumerkey"];
+        options.ConsumerSecret = Configuration["auth:twitter:consumersecret"];
+    });
+    ```
+
+### Multiple Authentication Schemes
+If your 1.x application has configured multiple authentication schemes, the default scheme needs to be identified differently in 2.0. 
+
+In 1.x, the `AutomaticAuthenticate` and `AutomaticChallenge` properties were intended to be set on a single authentication scheme. There was no good way to enforce this.
+
+In 2.0, these two properties have been removed as flags on the individual `AuthenticationOptions` instance and have moved into the base [AuthenticationOptions](/api/microsoft.aspnetcore.builder.authenticationoptions) class. The properties can be configured in the `AddAuthentication` method call within the `ConfigureServices` method of *Startup.cs*:
+
+```csharp
+services.AddAuthentication(CookieAuthenticationDefaults.AuthenticationScheme);
+```
+
+Alternatively, use an overloaded version of the `AddAuthentication` method to set more than one property:
+
+```csharp
+services.AddAuthentication(options =>
+{
+  options.DefaultScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+  options.DefaultChallengeScheme = OpenIdConnectDefaults.AuthenticationScheme;
+}).AddCookies().AddOpenIdConnect(options => { ... });
+```
+
+In this overloaded method example, the default scheme is set to `CookieAuthenticationDefaults.AuthenticationScheme`. The authentication scheme may alternatively be specified within your individual `[Authorize]` attributes or authorization policies.
+
+Define a default scheme in 2.0 if you want the user to be automatically signed in. An exception to that rule is the `AddIdentity` method. This method adds cookies for you and sets the default authenticate and challenge schemes to the application cookie `IdentityConstants.ApplicationScheme`. Additionally, it sets the default sign-in scheme to the external cookie `IdentityConstants.ExternalScheme`.
```

Comment:
... if you want the user to be automatically signed in or you use the `[Authorize]` attribute or authorization policies without specifying schemes.